### PR TITLE
Add CRD to configure droplogs

### PR DESCRIPTION
**Pull request by tanyatukade**
Signed-off-by: Tanya Tukade tanyatukade.123@gmail.com

### CODE REVIEW

File: provision/acc_provision/templates/aci-containers.yaml [hunk 1]
```diff
@@ -680,6 +680,64 @@ spec:
...
+                      type: string
+                    srcMAC:
+                      type: string
+                    destMAC:
+                      type: string
+                    srcPort:
+                      type: integer
+                    destPort:
+                      type: integer
+                    ipProto:
```
Comment (shastrinator) [inline provision/acc_provision/templates/aci-containers.yaml:738]: ipProto: is read as an integer from opflex.

Comment (tanyatukade) [reply provision/acc_provision/templates/aci-containers.yaml:739]: update it

File: provision/acc_provision/templates/aci-containers.yaml [hunk 2]
```diff
@@ -1306,6 +1365,17 @@ rules:
...
   - patch
+- apiGroups:
+  - "aci.droplog"
+  resources:
+  - droplogs
+  verbs:
+  - list
+  - watch
+  - get
+  - create
```
Comment (tomflynn) [inline provision/acc_provision/templates/aci-containers.yaml:1376]: Does the hostagent really need create/update/delete permissions for droplogs? I know our other CRDs are configured this way now but list/watch/get seems like all that's needed

File: provision/acc_provision/templates/aci-containers.yaml [hunk 3]
```diff
@@ -680,6 +680,65 @@ spec:
                     minimum: 1
                     maximum: 1023
 ---
+apiVersion: apiextensions.k8s.io/v1
```
Comment (shastrinator) [inline provision/acc_provision/templates/aci-containers.yaml:683]: General comment: There shoudl be only one CR that does enable/disable of droplogs. If this is not possible with CRD and web-hook, configmap may be a better option

File: provision/acc_provision/templates/aci-containers.yaml (hunk 2 above)
Comment (tanyatukade) [reply provision/acc_provision/templates/aci-containers.yaml:1376]: Yes, list/watch/get should be enough. Will update that, thanks!

File: provision/acc_provision/templates/aci-containers.yaml (hunk 3 above)
Comment (tanyatukade) [reply provision/acc_provision/templates/aci-containers.yaml:683]: As discussed, we will have a CRD with node selector and selectively enable droplog. We keep track of all CRs defined and then merge. If any CR has the feature disabled, then it’s disabled irrespective of other CRs.

**Comment by tanyatukade** [comment]
As discussed, we will have two separate CRDs. One CRD is used to enable/disable drop logging feature using node selectors. And the other CRD will have filters. I will update this PR of kind "DropLogEnable" and remove the filters as this will only used to enable/disable. Will create a new PR shortly for the 2nd CRD of kind "DropLogPruning".

**Comment by tanyatukade** [comment]
Updated the PR with two CRDs

**Review by shastrinator — DISMISSED** [review]

**Comment by tanyatukade** [comment]
Resolving conflicts

**Review by tomflynn — APPROVED** [review]

**Pull request merged**